### PR TITLE
Ocd 1503 - Updated to support test functionalities tied to practice type 

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/TestFunctionalityEntity.java [hunk 1]
```diff
@@ -36,6 +36,10 @@
     @JoinColumn(name = "certification_edition_id", insertable = false, updatable = false)
     private CertificationEditionEntity certificationEdition;
 
+    @OneToOne(optional = true, fetch = FetchType.EAGER)
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/TestFunctionalityEntity.java]: I don't think I've seen "EAGER" before. Most of the others use "LAZY". What's the difference? Why one or the other?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/certifiedProduct/CertifiedProduct2014Validator.java [hunk 2]
```diff
@@ -219,6 +246,20 @@ public void validate(final PendingCertifiedProductDTO product) {
...
+                pt.setId(product.getPracticeTypeId());
+                pt.setName(product.getPracticeType());
+
+                product.getErrorMessages().addAll(
+                        getPendingTestFunctionalityValidationErrors(
+                                EDITION_ID,
+                                pt,
+                                product.getCertificationCriterion()));
+            } catch (EntityRetrievalException e) {
+                //TODO - what to do here???
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/certifiedProduct/CertifiedProduct2014Validator.java]: I think we usually just throw it, and shrug.

File: chpl/chpl-service/src/main/resources/errors.properties [hunk 3]
```diff
@@ -139,6 +139,7 @@ listing.criteria.badParticipantProfessionalExperienceMonths=Professional Experie
 listing.criteria.badParticipantProductExperienceMonths=Product Experience Months is required for participant %s.
 listing.criteria.badParticipantComputerExperienceMonths=Computer Experience Months is required for participant %s.
 listing.criteria.fuzzyMatch=Changed %s for criteria %s from %s to %s.
+listing.criteria.testFunctionalityMisMatch=In Criteria %s, Test Functionality %s is not valid for Practice Type %s.
```
Comment (andlar) [inline chpl/chpl-service/src/main/resources/errors.properties]: There was a request to add "...something is for something settings only and is not valid...", right?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/TestFunctionalityEntity.java (hunk 1 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/TestFunctionalityEntity.java]: I usually do Lazy fetches and wherever the test functionality entity is being gotten in the dao i'll add "join fetch" code to also get the practiceType entity. I don't like hibernate doing extra queries behind the scenes which I think is what Eager fetches have the potential to do?

**Pull request merged**